packages/components/src/modifiers/hds-code-editor.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { assert } from '@ember/debug';
+import Modifier from 'ember-modifier';
+import {
+  lineNumbers,
+  highlightActiveLineGutter,
+  highlightSpecialChars,
+  drawSelection,
+  highlightActiveLine,
+  EditorView,
+  keymap,
+} from '@codemirror/view';
+import { EditorState, type Extension } from '@codemirror/state';
+import { javascript } from '@codemirror/lang-javascript';
+import { json } from '@codemirror/lang-json';
+import { sql } from '@codemirror/lang-sql';
+import { go } from '@codemirror/lang-go';
```

Comment:
suggestion: investigate loading languages asynchronously, to reduce the size of loaded assets for CodeEditor instances

Reply:
Good call. In the latest version we only load the editor code and the language package when the code editor is in-view. We also only load the language if one is specified.